wgpu-core/src/device/mod.rs

```diff
@@ -1483,6 +1483,7 @@ impl<B: GfxBackend> Device<B> {
 
         // `BTreeMap` has ordered bindings as keys, which allows us to coalesce
         // the descriptor writes into a single transaction.
+        let mut desc_set; // early declaration so it's dropped before write_map
```

Comment:
This comment is inverted - it gets dropped after write_map when unwinding.

What was the error in the first place?

Reply:
Eh, right, the comment is the wrong way around 😅 

Build error is
```
error[E0597]: `desc_set` does not live long enough
    --> wgpu-core/src/device/mod.rs:1792:26
     |
1792 |                     set: desc_set.raw_mut(),
     |                          ^^^^^^^^^^^^^^^^^^ borrowed value does not live long enough
...
1813 |     }
     |     -
     |     |
     |     `desc_set` dropped here while still borrowed
     |     borrow might be used here, when `write_map` is dropped and runs the `Drop` code for type `BTreeMap`
     |
     = note: values in a scope are dropped in the opposite order they are defined
```

The reason is the PR mentioned in the title, a soundness fix for `BTreeMap` which previously behaved (to dropck) as if no `K` and `V` values would be dropped when a `BTreeMap<K, B>` is dropped.